lib/cn/kvset.c

```diff
@@ -832,38 +775,27 @@ merr_t
 kvset_open(struct cn_tree *tree, uint64_t kvsetid, struct kvset_meta *km, struct kvset **ks)
 {
     merr_t         err;
-    uint           n_vblks = km->km_vblk_list.n_blks;
+    uint32_t       idc = km->km_vblk_list.n_blks;
+    uint64_t      *idv = km->km_vblk_list.blks;
     struct mbset * vbset = 0;
     struct mbset **vbsetv = &vbset;
     uint           vbsetc = 0;
     uint           len = 0;
 
-    if (n_vblks) {
-        u64 bufv[64];
-        u64 *idv;
-
-        idv = blkid_list_to_vec(&km->km_vblk_list, NELEM(bufv), bufv);
-        if (ev(!idv))
-            return merr(ENOMEM);
-
-        err = mbset_create(cn_tree_get_mp(tree), n_vblks, idv, sizeof(struct vblock_desc),
+    if (idc) {
+        err = mbset_create(cn_tree_get_mp(tree), idc, idv, sizeof(struct vblock_desc),
```

Comment:
Now we can stop creating temporary arrays of mblock IDs necessitated by mismatched APIs.